crates/network-scanner/src/ping.rs

```diff
@@ -0,0 +1,54 @@
+use std::net::Ipv4Addr;
+
+use network_scanner_net::tokio_raw_socket::TokioRawSocketStream;
+use network_scanner_proto::icmp_v4;
+
+use crate::NetworkScanError;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+type NetowrkScanResult<T> = Result<T, NetworkScanError>;
+
+pub async fn ping(ip: impl Into<Ipv4Addr>) -> NetowrkScanResult<()> {
```

Comment:
issue: This will cause the whole function to be inlined and compiled in the consumer crate.

suggestion: You can use the trick [mentioned here](https://github.com/Devolutions/IronRDP/blob/33c66edfc6ddc8fd47d83b3ef4ddf9b1875a4bc2/STYLE.md#avoid-monomorphization). It’s also fine for this function to not be generic at all given that we’ll not use it more than once. At you preference.

Reply:
updated